src/Engine/ProtoCore/Utils/ArrayUtils.cs

```diff
@@ -117,25 +118,43 @@ public static ClassNode GetGreatestCommonSubclassForArray(StackValue array, Runt
             return runtimeCore.DSExecutable.classTable.ClassNodes[orderedTypes.First()];
         }
 
-        public static Dictionary<int, StackValue> GetTypeExamplesForLayer(StackValue array, RuntimeCore runtimeCore)
+        public static IEnumerable<StackValue> GetTypeExamplesForLayer(StackValue array, RuntimeCore runtimeCore)
```

Comment:
this is an API break I believe.

Reply:
a Dictionary is an `System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<TKey,TValue>>` not general Ienumerable of StackValue

Reply:
I would really appreciate a summary for this method too 😈

Reply:
@mjkkirschner the problem with `GetTypeExamplesForLayer` was that it returned with only 1 `ArrayPtr` even if there were `[ArrayPtr, ArrayPtr]` but the 2 array pointers can be different. It will always return with the first one.

So it will fail for such cases: `[[1,2,3], [{dict}]]` and it will fail for `[[dict], [[dict]]]` for the same reason, where `dict` is any dictionary.